### PR TITLE
[IMM32_APITEST] Add JapanImeConvTestA/W testcases

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
Add IME conversion tests to verify IMM/IME support.
JIRA issue: [CORE-11700](https://jira.reactos.org/browse/CORE-11700)

## Proposed changes

- Add File `JapanImeConvTestA.c` and `JapanImeConvTestA` testcase.
- Add File `JapanImeConvTestW.c` and `JapanImeConvTestW` testcase.
- Add `JapanImeConvTest.h` and `resource.rc` files.

## Screenshots

Japanese WinXP (using preinstalled IME):
WinXP-pre
Successful.

Japanese WinXP (using `katahiromz/mzimeja`):
WinXP-mzime
Successful.

Japanese Win10 (using preinstalled IME):
Win10
Successful.

### CODE REVIEW

File: modules/rostests/apitests/imm32/JapanImeConvTest.h [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+ */
+
+#include <ntstatus.h>
+#define WIN32_NO_STATUS
+#include <windows.h>
+#include <windowsx.h>
+#include <imm.h>
+//#include <ddk/immdev.h>
+#include <wine/test.h>
+//#include <pseh/pseh2.h>
```
Comment (HBelusca) [inline modules/rostests/apitests/imm32/JapanImeConvTest.h:15]: Looks unnecessary at the moment.

File: modules/rostests/apitests/imm32/JapanImeConvTest.h [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+#include <stdio.h>
```
Comment (HBelusca) [inline modules/rostests/apitests/imm32/JapanImeConvTest.h:16]: Move before ntstatus and other inclusions.

File: modules/rostests/apitests/imm32/JapanImeConvTest.h [hunk 3]
```diff
@@ -0,0 +1,259 @@
+/*
+ * PROJECT:     ReactOS api tests
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Test for Japanese IME conversion
+ * COPYRIGHT:   Copyright 2022 Katayama Hirofumi MZ (katayama.hirofumi.mz@gmail.com)
+ */
+
+#include <ntstatus.h>
```
Comment (HBelusca) [inline modules/rostests/apitests/imm32/JapanImeConvTest.h:8]: For which thing is this needed? You don't seem to use any NTSTATUS values (at least explicitly) there.

File: modules/rostests/apitests/imm32/JapanImeConvTest.h [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+                ok(hIMC != NULL, "hIMC was NULL\n");
+                ++s_cWM_IME_ENDCOMPOSITION;
+
+                pszResult = (LPTSTR)calloc(cbResult + 1, sizeof(TCHAR));
+                ok(pszResult != NULL, "pszResult was NULL\n");
+                ImmGetCompositionString(hIMC, GCS_RESULTSTR, pszResult, cbResult);
+#ifdef UNICODE
+                printf("%s\n", WideToAnsi(CP_ACP, (LPTSTR)pszResult));
+#else
+                printf("%s\n", (LPTSTR)pszResult);
```
Comment (HBelusca) [inline modules/rostests/apitests/imm32/JapanImeConvTest.h:92]: Not using TRACE or something similar?

**Review by HBelusca — COMMENTED** [review]
What's missing somewhere in the JapanImeConvTest.h file, is an explanation of the idea of what the test is trying to do.

File: modules/rostests/apitests/imm32/JapanImeConvTest.h (hunk 1 above)
Comment (katahiromz) [reply modules/rostests/apitests/imm32/JapanImeConvTest.h]: Fixed.

File: modules/rostests/apitests/imm32/JapanImeConvTest.h (hunk 2 above)
Comment (katahiromz) [reply modules/rostests/apitests/imm32/JapanImeConvTest.h]: Deleted.

File: modules/rostests/apitests/imm32/JapanImeConvTest.h (hunk 3 above)
Comment (katahiromz) [reply modules/rostests/apitests/imm32/JapanImeConvTest.h]: Deleted.

File: modules/rostests/apitests/imm32/JapanImeConvTest.h (hunk 4 above)
Comment (katahiromz) [reply modules/rostests/apitests/imm32/JapanImeConvTest.h]: Used `trace`.

**Comment by katahiromz** [comment]
Now comments added.

**Review by HBelusca — APPROVED** [review]

**Comment by HBelusca** [comment]
How much does this test succeeds (or fails) currently in ReactOS?

**Comment by katahiromz** [comment]
Japanese without IME:
without-ime

Japanese with IME:
with-ime

Failing at `SM_IMMENABLED`.

**Comment by katahiromz** [comment]
Japanese ReactOS with IME (`katahiromz/mzimeja`):

It works now. Thank you!

**Pull request merged**